### DISCUSSION TITLE
Improve "on" directive documentation

### DISCUSSION THREAD

**Discussion by Angel-Ponce** [alpinejs/alpine#4306] [4. General]
I'm currently working with AlpineJS, and I tried to put `x-on:click` in some of my html elements, but, this not working as well, so, searching a little bit in google I found that the. `x-on` directive not works if `x-data` it's not present.

This "condiction" it's not present in the official documentation of `x-on` directive.

**Comment by SimoTod** [comment]
It's not just x-on, it's the same for the other directives: https://alpinejs.dev/start-here

If you just need a click handler without an alpine component you can just use the native https://www.w3schools.com/TAgs/att_onclick.asp

**Reply by Angel-Ponce** [reply]
Yeah! The main problem it's not about the use of AlpineJS or Native JavaScript

I'll appreciate if the documentation of directives remember me to take care when using AlpineJS features without `x-data` directive

**Comment by ekwoka** [comment]
Y'know, I'll just bite the bullet and PR a little note box on every directive page next to the first example.

And update the started guides to be more clear about it as well.

I don't really think it should be necessary, but this is just about self-preservation at this point 😆 😭

**Reply by Angel-Ponce** [reply]
What do you mean with "self-preservation"? 😶

**Reply by ekwoka** [reply]
To not have to deal with these questions anymore 😁
